### ISSUE TITLE
Make a forum for TF addons

### ISSUE THREAD

**Issue by gabrieldemarmiesse** [tensorflow/addons#1641]
**Describe the feature and the current behavior/state.**

Currently, when people have questions about using a feature of TF addons, they either go to issues, go to stackoverflow or the mailing list. Neither of them have the UX of the pytorch forums for example. As such I believe that we should have a proper forum. It would also be a nice proof of concept for the TF org as other project could later copy the work we do to create a forum.

**Relevant information**
- Are you willing to contribute it (yes/no): Yes, some of it
- Are you willing to maintain it going forward? (yes/no): yes, some of it
- Is there a relevant academic paper? (if so, where): no LOL
- Is there already an implementation in another framework? (if so, where): pytorch
- Was it part of tf.contrib? (if so, where): No

**Which API type would this fall under (layer, metric, optimizer, etc.)**

PR

**Who will benefit with this feature?**

All users!

**Any other info.**

We can have free google cloud credits if we fill a free form google docs explaining the value proposition. So we can host stuff if needed.

@seanpmorgan 
@AakashKumarNain 
@bhack

This issue can be used to drop ideas, implementations, whatever you think is relevant about how to setup the forum.

**Comment by AakashKumarNain** [comment]
Yes, yes yes! @gabrieldemarmiesse not only PyTorch but `mxnet` and `fastai` also have dedicated forums. Also, being a Kaggler, I am pretty sure that forums are very very useful for discussions regarding features, asking questions, aggregating discussions on a particular topic, etc.

**Comment by bhack** [comment]
I like it but honestly I will prefer to have a common space for all the SIGs cause it will really help to minimize accounts and help  cross-mentions on threads. 
See https://github.com/tensorflow/community/issues/204

Can you check if google is in the condition to decide about this?

If the response is negative we could start alone.

**Comment by gabrieldemarmiesse** [comment]
> I like it but honestly I will prefer to have a common space for all the SIGs cause it will really help to minimize accounts and help cross-mentions on threads.

That'd be the best. 

> Can you check if google is in the condition to decide about this?

Sadly I believe that at this point in time, if we wait for an official answer from google (even negative), it can take several months.

> If the response is negative we could start alone.

That'd be the best. I believe that google will follow-up on what we do if it works well. Should they decide to embrace it later, they have the manpower to transfer the discussions to a new organized forum. 

Right now we need to act as a POC and show that we add value.

What software/services do you guys know/recommend?

**Comment by bhack** [comment]
Sadly... I mentioned some tools in tensorflow/community#204

**Comment by gabrieldemarmiesse** [comment]
I believe we should start with one tool first to avoid having too much work to set it up. Should we start with discord or discourse?

**Comment by bhack** [comment]
Ok for discourse. 

Instead for discord, yes It Is quite popular like slack etc.. but If we prefer an opensource ecosystem I will go in the Mozilla direction with Riot/Matrix

**Comment by gabrieldemarmiesse** [comment]
@bhack would you be willing to follow-up on this and setup the discourse forum? If you need anything, you can ask the addons maintainers in this issue of course. I have no experience with discourse but if you need some stuff like hosting, we'll try to make it work.

**Comment by bhack** [comment]
Is the Opensource free hosting enough for our traffic?

https://blog.discourse.org/2018/11/free-hosting-for-open-source-v2/

**Comment by gabrieldemarmiesse** [comment]
I think so, if we need to upgrade, then it means we're successful and we can ask google to help us there :)

**Comment by bhack** [comment]
Can you submit the request at https://free.discourse.group/?

**Comment by gabrieldemarmiesse** [comment]
Could you do it? You can be one of the staff members of course, that will definitly help :)

**Comment by bhack** [comment]
Ok

**Comment by gabrieldemarmiesse** [comment]
Thanks a lot!

**Comment by bhack** [comment]
Done. We need to wait 2 business days to know if we are approved.

**Comment by gabrieldemarmiesse** [comment]
That's awesome, thanks!

**Comment by bhack** [comment]
In the meantime we have a realtime room on matrix at https://riot.im/app/#/room/#tensorflow/sig-addons:matrix.org.
We could bridge with the current Gitter `tensorflow-sig-addons` if you like.

**Comment by bhack** [comment]
@gabrieldemarmiesse @seanpmorgan If you like it I will give you the co-maitainership when you will join..

**Comment by gabrieldemarmiesse** [comment]
@bhack could you deactivate the gitter bot so that we can try riot without worrying about  posting messages on gitter?

**Comment by bhack** [comment]
Yes

**Comment by gabrieldemarmiesse** [comment]
Nice thanks!

**Comment by bhack** [comment]
Done, Is disconnected now.

**Comment by bhack** [comment]
@ewilderj as i see in the [community logos](https://github.com/tensorflow/community/blob/master/sigs/logos/README.md) folder can you add in the repository an [ICON version](https://www.tensorflow.org/extras/tensorflow_brand_guidelines.pdf) of the logo for this SIG (probably could be useful also for other SIGs)? Thanks

**Comment by bhack** [comment]
It Is ready. https://tfaddons.discourse.group
We can personalize graphics, theme and text.

When we are ready to announce these new tools in mailing lists and PR in Readme we could close this ticket

**Comment by gabrieldemarmiesse** [comment]
That's awesome, thanks @bhack ! We have to decide then what to do with the channels we already have: mailing list and gitter. Should we drop them?

**Comment by bhack** [comment]
I think that Is ok. When we are satisfued with the layout settings we could put the old channels in readonly mode e.g. one week after the announcement.

**Comment by ewilderj** [comment]
Whoa folks. Firstly I'm really glad that you want to move on this, but going from issue to creation in 3 days without wider consultation is a bit hasty. As demonstrated by the icon ping, you definitely know how to get a hold of me and chat anytime :) Also, SIG Addons has official Google contacts you can talk with.

We have some real constraints about forums and project-wide desires about how to manage interaction. I know it seems like it might be slower, but we'd like to collect and understand the user needs around TF Addons support so we ensure they're met in a larger way. We don't want to fragment the TensorFlow support space -- Addons users are also TensorFlow users in the large.

Can you follow up with me at ewj@google.com ?

cc/ @seanpmorgan @theadactyl

**Comment by gabrieldemarmiesse** [comment]
@ewilderj , could we follow-up in this issue? I believe that the community would benefit from this discussion as it would allow anyone to understand the decisions behind the tools we use in Addons.

**Comment by bhack** [comment]
@ewilderj My original opinion was expressed in this ticket at https://github.com/tensorflow/addons/issues/1641#issuecomment-612213964. 
Instead the original community issue, that I've mentioned in that comment, was stalled by more then two months as many other community issues.

Personally I understand that internally you need months of discussions, I know this and it is not a problem. I don't wont to re-engineer google pipelines/process :) 

The main problem IMHO and this is quite general about the relationship is that it is very hard to have a jointly thread between internal and 
contributors. 
Generally my experience is that when we start a thread this bootstrap an internal long thread without any external visibility or work-in-progess feebacks (e.g. this is a very important and  a good about Org example https://github.com/tensorflow/community/issues/29). :wink: 

It is not a problem for me to put down the infra. Let me know if you want to discuss this in private.

**Comment by ewilderj** [comment]
Thanks, and I appreciate the difficulty of this. TensorFlow is a unique and tricky project in many ways, and this is also one of them :)  The general problem is that each SIG is one of many, and all share the TensorFlow name, so we want to ensure users get a consistent and non-confusing approach.

We can definitely do better than months of apparent silence while internal threads go on, and we would like to share with you plans and some of our constraints around solving for the forum issue. Thea will follow up.

**Comment by bhack** [comment]
@ewilderj It could be unique and tricky as you want but transparency it about your policy as Org and anything else. If there is no organizational and corporate policy effort to enforce these "positive" behaviors not much can be done as we are not a foundation backing an opensource project.

I honestly appreciate more, sometimes contradictory discussions, on [MLIR's channels](https://mlir.llvm.org/) cause it is healthy for a community ecosystem.

**Comment by bhack** [comment]
> The general problem is that each SIG is one of many, and all share the TensorFlow name, so we want to ensure users get a consistent and non-confusing approach.

I was manly interested in Addons but as I agree with this part of your point of view I've tried to go months ago through https://github.com/tensorflow/community/issues/204 also after many not positive experience on using the "community repo routing".

Also this is an ideal theoretic goal but as you see you have not enforced this cause we have approved PRs like https://github.com/tensorflow/community/pull/41.

Also we have Keras that I don't understand, now that it is completely standalone, can do things that other SIGs cannot do. It have both Gitter and Slack official channels.

It is hard to uniform stuffs if we are P2P reality but we don't have a strong "P2P protocol" that can generate consensus.

My just 2c abut this initiative is that  we have proposed completely opensource tools and protocol that could be self hosted if needed. So any vendor lock-in.

**Comment by bhack** [comment]
My bad on https://github.com/tensorflow/community/pull/41 cause was not merged. But on Keras was correct also if it is probably not a SIG anymore.

**Comment by theadactyl** [comment]
@bhack @seanpmorgan 

Multiple thoughts here. One is that /community admins need to triage issues more regularly. :) But please feel free to @ me in the future into any issue in this repo you're not seeing a timely response on.

Another is that we may need to have a clearer process around creating new community processes, as this shouldn't be done on the fly to make sure our communications channels have the support and resources they need to thrive. Issue here: https://github.com/tensorflow/community/issues/225

Finally, very glad to see the enthusiasm for forums! This is something we're actively working on internally. It turns out that the implementation piece is a lot more challenging than we originally foresaw in order to set things up properly, safely, and in a way that best facilitates both Tensorflow team and community engagement and leadership. I'll work with @joanafilipa to get more updates out to the SIG leads on this effort and how they can help. 

In the meantime, we'd appreciate holding off on setting up this particular forum -- amongst other concerns, I think a forum platform should serve multiple TF SIGs/projects to avoid proliferation of comms channels and the administration challenges that could introduce.

**Comment by bhack** [comment]
> This is something we're actively working on internally. It turns out that the implementation piece is a lot more challenging than we originally foresaw in order to set things up properly, safely, and in a way that best facilitates both the Tensorflow team's and the community's engagement and leadership. I'll work with @joanafilipa to get more updates out to the SIG leads on this effort and how they can help.

@theadactyl Thank you for the update. Can that thread be transparent and so a public topic for the SIG communities or it is an internal one related the Company internal implications about adopting these channels?

Cause LLVM-MLIR, Pytorch, Mxnet etc.. are on discourse. Why Tensorflow it so special? 
And if it is so special can we discuss in public why this is always the case (just to mention again "TensorFlow is a unique and tricky project in many ways")?

**Comment by gabrieldemarmiesse** [comment]
@theadactyl thanks a lot for your help and insights! That's definitely good to have a point of contact when community members outside google would like to propose changes to tools.

For the sake of community visibility and information, 

* what is the rational behind creating a custom forum when open sources options are available?
* What is the timeframe concerning the availability this cutom made forum? If it's 3-4 months, that's ok, if it's more, maybe using a free, open-source option in the meantime would ensure a good UX. This good UX for several months would then be worth the migration costs.

**Comment by theadactyl** [comment]
@bhack @gabrieldemarmiesse 
> @theadactyl Thank you for the update. Can that thread be transparent and so a public topic for the SIG communities? Or is it an internal one related to the Company internal implications about adopting these channels?
This is an involved discussion that we can't have publicly. What we can do is provide more transparency on where we are with the process. Right now we are exploring multiple options, including existing OSS forums. Event then, there are implementation details that are important to get right and that take time to consider. 

This thread itself was very helpful in confirming that these forums are wanted and needed, and that the SIG community would like to be involved in their formation. We look forward to sharing more with you as soon as possible.

**Comment by bhack** [comment]
@gabrieldemarmiesse @seanpmorgan I've closed Discourse and Matrix channel resources since these things are evaluated through an internal Google process where we have no visibility.

I suggest also to close this ticket because we is no-sense to deal with non-public topics on a public github repo.

**Comment by theadactyl** [comment]
To be clear, there's no need to shut down any chats platforms that are
already being used for SIGs.

**Comment by bhack** [comment]
@theadactyl Yes we just closed the new ones of this issue.

**Comment by gabrieldemarmiesse** [comment]
Closing as we cannot do much about it.

**Comment by bhack** [comment]
@theadactyl @ewilderj:
> In practice, the way this is happening is that Element (the company founded by the Matrix core team to fund Matrix development) is acquiring Gitter from GitLab

https://matrix.org/blog/2020/09/30/welcoming-gitter-to-matrix
https://element.io/blog/gitter-is-joining-element/

**Comment by bhack** [comment]
Finally we have an official TF forum for the whole ecosystem on Discourse https://discuss.tensorflow.org/

**Comment by ewilderj** [comment]
I'm so glad that we do, and thanks everyone for the patience and engagement! The machinery involved in ensuring we do this well for an entire project that has this many stakeholders, and taking trust, safety and moderation into account, has been pretty immense. Kudos to @theadactyl and @joanafilipa for leading this project through.
